Update docs and code to remove GA3 references and match GA4 reality (#14910)
mozilla/bedrock: docs/attribution/0001-analytics.rst

Mozorg analytics
=================

Google Analytics
****************

We use Google Analytics (GA) to track how our website is used.

*Implementation*

Google calls a 3rd party JavaScript library that is imported by adding a script tag to a website,

> Google names these "tag"s.

Reviewer: ```suggestion
Google names these "tags".
```

Author: ✅

We do no not use any of GTM's most powerful features. Our use of GTM is a historical artifact of our
Universal Analytics (UA) (aka GA3) implementation. GTM is a tool to let non-technical users add code to a website, however,
there is no one outside of the websites team who is trained to do that.

The only tag we include with GTM is the Google Tag (GTag), and that in turn connects to our Google Analytics (GA) account.

There are three ways we send events from our site to GA:

1. Google's Enhanced Event Measurement logs certain events automatically.
2. GTM watches elements with specific data-attributes for user interaction
3. Bedrock watches for specific events, formats them, and sends them to GTM using the dataLayer

*Implementation Principles*

The current implementation tries very hard to keep any kind of logic and formatting of our events in bedrock where it can be
tested, code reviewed, and version controlled.

Many user interactions will trigger multiple events. For example: clicking the "get subscription" button on the VPN page will
trigger ``link``, ``cta_click``, and ``begin_checkout`` events. This is totally fine.

> If you're creating a new event use the two word noun_verb scheme that Google came up with and use snake_case,

Reviewer: I would italicize these

```suggestion
If you're creating a new event, use the two word _noun_verb_ scheme that Google came up with and use _snake_case_,
```

even if one of your noun or verb is more than one word.

*Debugging*

- GTM is managed with Tag Manager https://tagmanager.google.com/
- Our gTag is also managed with Tag Manager https://tagmanager.google.com/#/home#tags

> - GA, GTM, and gTag can be debugged using Tag Assistant https://tagassistant.google.com/

Reviewer: ```suggestion
- GA, GTM, and GTag can be debugged using Tag Assistant https://tagassistant.google.com/
```

- GA also has a debug view https://support.google.com/analytics/answer/7201382?hl=en

How can visitors opt out of GA?
-------------------------------

Visitors to the website can opt-out of loading Google Analytics on our
website by enabling `Do Not Track (DNT)`_ in their web browser. We
facilitate this by using a `DNT helper`_ that our team maintains. We
also respect `Global Privacy Control (GPC)`_ as an additional opt-out
signal.

Alternatively, visitors can also opt-out of GA by visiting the
`cookie settings page`_, which is linked to in the main site
footer on every page.

Event Tracking
--------------

Enhanced Event Measurement
~~~~~~~~~~~~~~~~~~~~~~~~~~

Pageviews, scrolls, video events, and outbound link clicks are being collected using GA4's `enhanced event measurement`_.

> Some form submissions are also being collected but newsletter signups are not. `(See Bug #13348)`_ They are instead

Reviewer: ```suggestion
Some form submissions are also being collected but newsletter signups are not `(see bug #13348)`_ . They are instead
```

Author: ✅

being tracked with the ``newsletter_subscribe`` event.

See the full list of events GA collects by default. https://support.google.com/analytics/answer/9216061

Data Attributes
~~~~~~~~~~~~~~~

One place where we do still rely on GTM to watch for an trigger events is when we use data-attributes to pass
information. There are only two events we do this with and they are primarily used on buttons and links.

CTA Click
"""""""""

CTA ("Call to Action") click is intented to track the one or two main actions the page is designed to get the user
to do. This data-attribute can be used on either ``<button>`` or ``<a>``. Links can go to or away from the site,
a button might trigger a JS function instead of going away from the page, that's still a cta.

> The attribute ``data-cta-text`` must be present to trigger the event. All links to accounts.firefox.com must also use cta-type.

Reviewer: ```suggestion
The attribute ``data-cta-text`` must be present to trigger the event. All links to accounts.mozilla.com must also use ``datacta-type`` .
```

+-----------------------+----------------------------------------------------------------------------------+
| Data Attribute        | Expected Value                                                                   |
+=======================+==================================================================================+

> | ``data-cta-text`` *   | Text or name of the link (e.g. ``Sign Up``, ``Start Here``, ``Get Relay``,       |

Reviewer: **Question:** What is this asterisks referring to?

Author: It's a required attribute. I added a note in the docs.

.. code-block:: html

    <a href="https://monitor.firefox.com/" data-cta-text="Check for breaches" data-cta-type="fxa-monitor">Check for breaches</a>

    <a href="{{ url('firefox.browsers.mobile.get-app') }}" data-cta-text="Send Link for Firefox Mobile" data-cta-position="banner">Send me a link</a>

    <a href="{{ url('firefox.browsers.mobile.ios') }}" data-cta-text="Firefox for iOS">Firefox for iOS</a>

Link Click
""""""""""

Link click is intended to track links that are of interest but not the focus of the page. Examples include links in paragraphs,
lists, FAQs, supplemental content, or in a navigation menu. Links can go to or away from the site.

The attribute ``data-link-text`` must be present to trigger the event.

+-----------------------+------------------------------------------------------------------------------------------------+
| Data Attribute        | Expected Value                                                                                 |
+=======================+================================================================================================+

> | ``data-link-text`` *  | Text or name of the link (e.g. ``Monitor``, ``Features``, ``Instagram (mozilla)``,             |

Reviewer: **Question:** What is this asterisks referring to?

Author: It's a required attribute. I added a note in the docs.
